CMakeLists.txt

```diff
@@ -806,6 +806,7 @@ add_library(extra INTERFACE)
 target_link_libraries(extra INTERFACE ${CMAKE_DL_LIBS})
 target_link_libraries(extra INTERFACE Boost::boost)
 target_link_libraries(extra INTERFACE Threads::Threads)
+target_link_libraries(extra INTERFACE Tracy::TracyClient)
```

Comment:
```suggestion
if(TRACY_ENABLE)
    target_link_libraries(extra INTERFACE Tracy::TracyClient)
endif()
```

Reply:
Tracy is meant to compile to nothing if `TRACY_ENABLE` is not set, setting the CMake flag defines the preprocessor definition `TRACY_ENABLE` which causes all the Tracy code in the header and implementation to be compiled out like: https://github.com/wolfpld/tracy/blob/4a5a21cdb08c3b554bf1857b4f2f33d7f7db0207/public/tracy/Tracy.hpp#L19 

We could add a shim `tracy.h` that conditionally includes the real `tracy.h` and or otherwise defines the Tracy macros out ourselves and remove the cost of building the thing from Oxen if you'd like?

Reply:
This was meant to go along with the later comment about avoiding the add_subdirectory, in which case the Tracy::TracyClient target wouldn't be available.

Reply:
I've done the change which is what I think you intended. Linking to `TracyClient` is what adds the include paths to allow usage of `<tracy/Tracy.hpp>` in the codebase. So getting rid of it then means I added our own `tracy_shim.h` which handles when `TracyClient` is _not_ linked and `<tracy/Tracy.hpp>` is not available.

I think this is better anyway since the build is already super complicated and we can avoid the meta build-steps incurred from including Tracy since we have a lot of build targets that may or may not work with it out of the box.